vsclib/vsc_test.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python3
+# (C) 2022 MBition GmbH
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+#
+
+import sys
+from vsclib import vsc
+
+def test_vsc():
+    root_ns = vsc.create_root_namespace()
+    a_ns = root_ns.add_namespace("a")
+    a_a_ns = a_ns.add_namespace("a")
+    a_b_ns = a_ns.add_namespace("b")
+    a_b_a_ns = a_b_ns.add_namespace("a")
+
+    # Add an int8 typedef at a.b.a
+    my_typedef_dt = a_b_a_ns.add_typedef("my_typedef", "int8")
+
+    # Mathing dictionary
+    my_typedef_dict = {
+        "name": "my_typedef",
+        "datatype": "int8"
+    }
+
+
+    # Locate datatype we just added via root search from root
+    if root_ns.find_datatype_by_name(".a.b.a.my_typedef") != my_typedef_dt:
+        print("Fail typedef root search test 1")
+        sys.exit(1)
+
+    print("Pass typedef root search test 1")
+
+    # Locate datatype we just added via root search from different child
+    if a_a_ns.find_datatype_by_name(".a.b.a.my_typedef") != my_typedef_dt:
+        print("Fail typedef root search test 2")
+        sys.exit(4)
```

Comment:
What is the pattern used for error codes? Just trying to be unique? if so should this one not be 2?